crates/red_knot_python_semantic/resources/mdtest/type_properties/is_disjoint_from.md

```diff
@@ -130,6 +131,29 @@ static_assert(not is_disjoint_from(Y, Z))
 static_assert(not is_disjoint_from(Intersection[X, Y], Z))
 static_assert(not is_disjoint_from(Intersection[X, Z], Y))
 static_assert(not is_disjoint_from(Intersection[Y, Z], X))
+
+# If one side has a positive fully-static element and the other side has a negative of that element, they are disjoint
+static_assert(is_disjoint_from(int, Not[int]))
+static_assert(is_disjoint_from(Intersection[X, Y, Not[Z]], Intersection[X, Z]))
+static_assert(is_disjoint_from(Intersection[X, Not[Literal[1]]], Literal[1]))
+
+class Parent: ...
+class Child(Parent): ...
+
+static_assert(not is_disjoint_from(Parent, Child))
+static_assert(not is_disjoint_from(Parent, Not[Child]))
+static_assert(not is_disjoint_from(Not[Parent], Not[Child]))
+static_assert(is_disjoint_from(Not[Parent], Child))
+static_assert(is_disjoint_from(Intersection[X, Not[Parent]], Child))
+static_assert(is_disjoint_from(Intersection[X, Not[Parent]], Intersection[X, Child]))
+
+static_assert(not is_disjoint_from(Intersection[Any, X], Intersection[Any, Not[Y]]))
+static_assert(not is_disjoint_from(Intersection[Any, Not[Y]], Intersection[Any, X]))
+
+static_assert(is_disjoint_from(Intersection[int, Any], Not[int]))
+static_assert(is_disjoint_from(Not[int], Intersection[int, Any]))
+
+# [quickcheck] TEST FAILED. Arguments: (Intersection { pos: [SubclassOfAbcClass("ABC"),Any], neg: [] }, Intersection { pos: [], neg: [SubclassOfAbcClass("ABC")] })
```

Comment:
Is this just a leftover comment that should be removed? I'm not seeing any failing quickcheck tests on the current version of the PR, and when I add this example as an mdtest, it passes (in both permutations). (It looks like the tests immediately above here are the same scenario, just using `int` in place of `abc.ABC`.)

Below is just for demonstration of exactly what I tested locally; we don't necessarily need to add these tests if we already have the equivalent above.

```suggestion
```

Reply:
Going to go ahead and remove this comment, then merge the PR. Let me know if I've missed something that was still TODO represented by this comment!